tests/unit/backend/corpora/common/entities/test_dataset_asset.py

```diff
@@ -0,0 +1,71 @@
+from backend.corpora.common.corpora_orm import (
+    DatasetArtifactType,
+    DatasetArtifactFileType,
+)
+from backend.corpora.common.entities.dataset import Dataset
+from backend.corpora.common.entities.dataset_asset import DatasetAsset
+from tests.unit.backend.corpora import CorporaTestCaseUsingMockAWS
+from tests.unit.backend.utils import BogusDatasetParams
+
+
+class TestDatasetAsset(CorporaTestCaseUsingMockAWS):
```

Comment:
I think there's a few negative tests missing here for when `generate_presigned_url` or `head_object` fails. You can use `patch` and `side_effect` to force an exception and verify that the DatasetAsset class catches it and returns None.

Reference: https://docs.python.org/3/library/unittest.mock.html

Reply:
we already test this for get_file_size. I add a mock test for generate_presigned_url